R/nzchar_linter.R

```diff
@@ -0,0 +1,103 @@
+#' Require usage of nzchar where appropriate
+#'
+#' [nzchar()] efficiently determines which of a vector of strings are empty
+#'   (i.e., are `""`). It should in most cases be used instead of
+#'   constructions like `string == ""` or `nchar(string) == 0`.
+#'
+#' One crucial difference is in the default handling of `NA_character_`, i.e.,
+#'   missing strings. `nzchar(NA_character_)` is `TRUE`, while `NA_character_ == ""`
+#'   and `nchar(NA_character_) == 0` are both `NA`.
+#'
+#' @evalRd rd_tags("nzchar_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+nzchar_linter <- function() {
+  # use string-length to capture both "" and ''
+  # if (any(x == "")) is not treated like it's part of if(), but
+  #   any(if (x == "") y else z) _is_ treated so. this condition looks for the
+  #   expr to be inside a call that's _not_ above an IF/WHILE.
+  comparison_xpath <- "
+  //STR_CONST[string-length(text()) = 2]
+    /parent::expr
+    /parent::expr[
+      (EQ or NE or LT or GT or LE or GE)
```

Comment:
Why lint `<, >, <=, >=`?

Reply:
Happens very rarely but does seem like it's done. Maybe as a little code golf thing?

https://github.com/search?q=lang%3AR+%2F%5B%5E%23%3D-%5D%3E%5Cs*%22%22%2F&type=code

Reply:
Interesting, there are even some hits for `<=`. Thanks for checking.